### PR TITLE
Load config

### PR DESCRIPTION
**Pull request by msimonin**
Review the way config is loaded.
Some cleaning on vagrant and g5k providers code.

### CODE REVIEW

File: enos/provider/g5k.py [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
 from netaddr import IPAddress, IPNetwork, IPSet
 from provider import Provider
-from ..utils.extra import build_resources, expand_topology, build_roles
 from ..utils.constants import EXTERNAL_IFACE
+from ..utils.extra import build_resources, expand_topology, build_roles
+from ..utils.provider import load_config
 
 import execo as EX
 import execo_g5k as EX5
+import execo_g5k.api_utils
```
Comment (rcherrueau) [inline enos/provider/g5k.py]: I expect some redundancy with line 2: `from execo_g5k import api_utils as api`.

File: enos/templates/Vagrantfile.j2 [hunk 2]
```diff
@@ -11,7 +11,7 @@ easy_install pip && ln -s /usr/local/bin/pip /usr/bin/pip || true
 SCRIPT
 
 Vagrant.configure(2) do |config|
-  config.vm.box = "debian/jessie64"
+  config.vm.box = "{{ provider_conf.box }}"
```
Comment (rcherrueau) [inline enos/templates/Vagrantfile.j2]: 👍

File: enos/tests/provider/test_g5k.py [hunk 3]
```diff
@@ -35,3 +38,106 @@ def test_not_enough_nodes_strict(self):
...
+
+
+class TestCreateReservation(unittest.TestCase):
+    def equalsJobSpecs(self, expected, actual):
+        # We only consider some attribute here, assuming it's enough
+        e = map(lambda (oar, site): (oar.resources, oar.name, site), expected)
+        a = map(lambda (oar, site): (oar.resources, oar.name, site), actual)
+        self.assertItemsEqual(e, a)
+
+    def test_1(self):
```
Comment (rcherrueau) [inline enos/tests/provider/test_g5k.py]: Could you change the name of this method to make feature tested explicit.

Comment (msimonin) [inline enos/tests/provider/test_g5k.py]: I removed this test (legacy code).

File: enos/provider/g5k.py (hunk 1 above)
Comment (msimonin) [inline enos/provider/g5k.py]: Right. I removed this line in favor a the previous one.

**Comment by msimonin** [comment]
Done.
Note that the travis build is failing because #113 isn't merged yet :)

**Pull request merged**